.testcoverage.yaml

```diff
@@ -0,0 +1,47 @@
+# (mandatory)
+# Path to coverprofile file (output of `go test -coverprofile` command)
+profile: build/test/coverage.out
+
+# (optional)
+# When specified reported file paths will not contain local prefix in the output
+local-prefix: "github.com/nginx/agent/v3"
+
+# Holds coverage thresholds percentages, values should be in range [0-100]
+threshold:
+  # (optional; default 0)
+  # The minimum coverage that each file should have
+  # file: 70
+
+  # (optional; default 0)
+  # The minimum coverage that each package should have
+  package: 80
+
+  # (optional; default 0)
+  # The minimum total coverage project should have
+  total: 80
+
+# Holds regexp rules which will override thresholds for matched files or packages using their paths.
+#
+# First rule from this list that matches file or package is going to apply new threshold to it.
+# If project has multiple rules that match same path, override rules should be listed in order from
+# specific to more general rules.
+#override:
+  # Increase coverage threshold to 100% for `foo` package (default is 80, as configured above)
+#  - threshold: 100
+#    path: ^pkg/lib/foo$
+
+# Holds regexp rules which will exclude matched files or packages from coverage statistics
+exclude:
+  # Exclude files or packages matching their paths
+  paths:
+    - \.pb\.go$       # Excludes all protobuf generated files
+    - \.gen\.go       # Excludes generated files
+    - ^fake_.*\.go    # Excludes fakes
+    - ^test/.*$
+    - app.go          # app.go and main.go should be tested by integration tests.
+    - main.go
+    - internal/datasource/config # These need to be addressed
```

Comment:
what needs to be addressed here?

Reply:
The package's coverage is < 80%, @aphralG is working on adding more unit tests to that dir I believe, so hoping this will get addressed there 🤞